stratum/hal/lib/barefoot/bf_pal_wrapper.cc

```diff
@@ -280,6 +281,23 @@ ::util::Status BFPalWrapper::PortLoopbackModeSet(int unit, uint32 port_id,
   return ::util::OkStatus();
 }
 
+::util::Status PortIdFromPortKeyGet(int unit, PortKey port_key,
+                                    uint32* sdk_port_id) {
+  std::string port_str = absl::StrFormat(
+      "%d/%d", port_key.port, port_key.channel);
```

Comment:
Note that the chassis config is using a 1-base channel number while SDE uses 0-base. 

In `stratum/hal/lib/common/utils.h` the description of the PortKey shows that there are three possible values of a port key:
non-zero: channelized
0: non-channelized
-1: the channel is not important at all

Reply:
good catch